### DISCUSSION TITLE
Use sphinx[test] for extension testing?

### DISCUSSION THREAD

**Discussion by jdillard** [sphinx-doc/sphinx#12333] [Q&A]
Is it advised to install `sphinx[test]` when testing extensions after this change https://github.com/sphinx-doc/sphinx/commit/66fa790b3a41b97ccbdd61e072cb2098886a7a74?

I was getting `ModuleNotFoundError: No module named 'defusedxml'` errors, but using `sphinx[test]` vs just `sphinx` in my tox `deps` fixed it.

I know the testing framework isn't well documented currently, but wondering if this should be added to the docs in the future if that is indeed the case.

**Comment by picnixz** [comment]
I prefer having that topic as an issue rather as a discussion, so I'm closing it. I've suggested having a `sphinx[dev]` *and* a `sphinx[test]` to make the distinction clearer (it might happen that Adam decides otherwise and we'll be back to a discussion but I feel that an issue is easier in this case).

See: https://github.com/sphinx-doc/sphinx/issues/12339
